meta-ostro/conf/distro/include/ostroproject-ci.inc

```diff
@@ -31,27 +31,86 @@ SDKMACHINE ?= "x86_64"
 # Generate source mirror tarballs:
 BB_GENERATE_MIRROR_TARBALLS = "1"
 
-# Include the devel settings
+# If (and only if) building in the CI, the BUILD_ID is guaranteed to
+# be consistent across different builds and therefore it can and
+# should be included in the os-release package.
+#
+# When doing the same outside of the CI, we'd end up with an
+# os-release file that has BUILD_ID = ${DATETIME} for the first time
+# that the os-release file was generated. It does not get re-generated
+# (explicit vardepsexclude in os-release.bb) because that would cause
+# constant rebuilds.
+OS_RELEASE_FIELDS_append = " BUILD_ID"
```

Comment:
On Thu, 2016-03-31 at 09:16 -0700, Alexander D. Kanevskiy wrote:
> Here is default set in OE-Core
> meta/recipes-core/os-release/os-release.bb:BUILD_ID ?= "${DATETIME}"

That default is useless, as discussed today with Joshua. It's not used
at all and using it would be problematic (for example, not consistent
between builds for different machines, which breaks the all-arch
os-release package).

> If it's not used for generating OS_VERSION, don't carry on old hack
> that @igor-stoppa added to get at least some predictable number in
> this field.
> 
> OS_VERSION is something that can stay even if swupd feature not
> enabled.
> Have it with some sensible defaults, like with swupd case, and with
> data from jenkins in ostroproject-ci.inc, but don't blindly copy&paste
> old lines that has no sense anymore.

I wasn't privy to the discussions you had with Igor. I'd prefer to have
the CI system set OS_VERSION, but as it doesn't do that at the moment,
the code still makes sense to me.

So what do you propose regarding OS_VERSION and BUILD_ID? Please be
specific, ensure that the CI is ready for it, and I'll implement as you
propose.